src/config/config.intern.h

```diff
@@ -0,0 +1,93 @@
+manipulate the configuration as a whole, e.g. protocol/helper, in order to communicate with other processes.
+
+The general-purpose functions for querying the current configuration are found in config.h.
+***********************************************************************************************************************************/
+#ifndef CONFIG_CONFIG_INTERN_H
+#define CONFIG_CONFIG_INTERN_H
+
+#include "config/config.h"
+
+/***********************************************************************************************************************************
+The maximum number of keys that an indexed option can have, e.g. pg256-path would be the maximum pg-path option
+***********************************************************************************************************************************/
+#define CFG_OPTION_KEY_MAX                                          256
+
+/***********************************************************************************************************************************
+Configuration data. These structures are not directly user-created or accessible. configParse() creates the structures and uses
+cfgInit() to load it as the current configuration. Various cfg*() functions provide access.
+***********************************************************************************************************************************/
+typedef struct ConfigOptionValue
+{
+    bool negate;                                                // Is the option negated?
+    bool reset;                                                 // Is the option reset?
+    unsigned int source;                                        // Where the option came from, i.e. ConfigSource enum
+    const Variant *value;                                       // Value
+} ConfigOptionValue;
+
+typedef struct Config
+{
+    MemContext *memContext;                                         // Mem context for config data
+
+    // Generally set by the command parser but can also be set by during execute to change commands, i.e. backup -> expire
+    ConfigCommand command;                                          // Current command
+    ConfigCommandRole commandRole;                                  // Current command role
+
+    String *exe;                                                    // Location of the executable
+    bool help;                                                      // Was help requested for the command?
+    StringList *paramList;                                          // Parameters passed to the command (if any)
+
+    // Group options that are related together to allow valid and test checks across all options in the group
+    struct
+    {
+        bool valid;                                                 // Is option group valid for the current command?
+        unsigned int indexTotal;                                    // Max index in option group
```

Comment:
I keep reading this `// Max index in option group` and getting the wrong impression. I think the comment maybe should read `Number of indexes with values in option group` - or something. The problem I have is I keep thinking it is 8 because that is the max index in the group but after reviewing the code, I see it is, say 2, which would be there are 2 values in the array which could be pg1 and pg8.

Reply:
Sure, that makes sense.

Reply:
fixed in 0bb8d05462c30300b9416b544829698dae11dc2c